src/PhysicsEngine.cpp

```diff
@@ -0,0 +1,119 @@
+#include "PhysicsEngine.h"
+#include "Entity.h"
+#include "AK/AK.h"
+
+void PhysicsEngine::initialize()
+{
+    auto const physics_engine = std::make_shared<PhysicsEngine>();
+    set_instance(physics_engine);
+}
+
+void PhysicsEngine::update_physics()
+{
+    solve_collisions();
+}
+
+void PhysicsEngine::emplace_collider(std::shared_ptr<Collider2D> const& collider)
+{
+    colliders.emplace_back(collider);
+}
+
+void PhysicsEngine::remove_collider(std::shared_ptr<Collider2D> const& collider)
+{
+    // Check if the shared pointer is already in the vector
+    auto const it = std::find_if(colliders.begin(), colliders.end(),
+        [&](const std::weak_ptr<Collider2D>& wp) {
+            return wp.lock() == collider;
+        });
+
+    if (it != colliders.end())
+        colliders.erase(it);
+}
```

Comment:
We try to use AK::swap_and_erase if the order of colliders in the vector does not matter.